assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -0,0 +1,561 @@
+
+        // Construct updated imageSource builder image object based on it's kind
+        var from = {};
+        if ($scope.options.pickedImageSourceType === "ImageStreamTag") {
+          from = {
+            kind: $scope.options.pickedImageSourceType,
+            namespace: $scope.options.pickedImageSourceNamespace,
+            name: $scope.options.pickedImageSourceImageStream + ":" + $scope.options.pickedImageSourceImageStreamTag
+          };
+        } else if ($scope.options.pickedImageSourceType === "DockerImage") {
+          from = {
+            kind: $scope.options.pickedImageSourceType,
+            name: $scope.options.pickedImageSourceDockerImage
+          };
+        }
+        $scope.updatedBuildConfig.spec.source.image.from = from;
+      }
+
+      // Construct updated builder image object based on it's kind
+      var from = {};
+      if ($scope.options.pickedBuildFromType === "ImageStreamTag") {
+        from = {
+          kind: $scope.options.pickedBuildFromType,
+          namespace: $scope.options.pickedBuildFromNamespace,
+          name: $scope.options.pickedBuildFromImageStream + ":" + $scope.options.pickedBuildFromImageStreamTag
+        };
+      } else if ($scope.options.pickedBuildFromType === "DockerImage") {
+        from = {
+          kind: $scope.options.pickedBuildFromType,
+          name: $scope.options.pickedBuildFromDockerImage
+        };
+      }
+      if ($scope.options.pickedBuildFromType === "None") {
+        delete $filter('buildStrategy')($scope.updatedBuildConfig).from
+      } else {
+        $filter('buildStrategy')($scope.updatedBuildConfig).from = from;
+      }
+
+      // Construct updated output image object based on it's kind
+      var to = {};
+      if ($scope.options.pickedPushToType === "ImageStreamTag") {
+        to = {
+          kind: $scope.options.pickedPushToType,
+          namespace: $scope.options.pickedPushToNamespace,
+          name: $scope.options.pickedPushToImageStream + ":" + $scope.options.pickedPushToImageStreamTag
+        };
+      } else if ($scope.options.pickedPushToType === "DockerImage") {
+        to = {
+          kind: $scope.options.pickedPushToType,
+          name: $scope.options.pickedPushToDockerImage
+        };
+      }
+      if ($scope.options.pickedPushToType === "None") {
+        // If user will change the output reference to 'None' shall the potential PushSecret be deleted as well?
+        // This case won't delete them.
+        delete($scope.updatedBuildConfig.spec.output.to)
+      } else {
+        $scope.updatedBuildConfig.spec.output.to = to;
+      }
+
+      // Update envVars
+      var updateEnvVars = [];
+      angular.forEach($scope.envVars, function(v, k) {
+        var env = {
+          name: k,
+          value: v
+        };
+        updateEnvVars.push(env);
+      });
+      $filter('buildStrategy')($scope.updatedBuildConfig).env = updateEnvVars;
+
+      // Update triggers
```

Comment:
@benjaminapetersen will decompose and put as much of the logic into a buildConfig service. Thanks Ben !